security/fma-l1-pectra.md

```diff
@@ -0,0 +1,94 @@
+# Upgrade 12 (L1 Pectra Defense): Failure Modes and Recovery Path Analysis
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+
+- [Introduction](#introduction)
+- [Generic Items](#generic-items)
+- [Audit Requirements](#audit-requirements)
+- [Action Items](#action-items)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+|                     |                      |
+| ------------------- | -------------------- |
+| Author              | George Knee          |
+| Created at          | 5th February 2025    |
+| Needs Approval From |                      |
+| Other Reviewers     |                      |
+| Status              | Draft                |
+
+## Introduction
+
+This document is intended to be shared in a public space for reviews and visibility purposes. It covers Upgrade 12, which involves the following changes:
+
+- (Consensus Layer) Ability to parse and validate blocks and receipts with EIP-7702 transactions, as well as blocks with a non-nil EIP-7685 requestsHash field. 
+- (Smart Contracts) For chains using Fault Proofs, L1 contracts updates which reference an updated absolute prestate hash
+
+Each change has its own section below with a list of Failure Modes.
+
+Below are references for this project:
+
+- [Github tracker](https://github.com/orgs/ethereum-optimism/projects/117/views/9)
+- [Specs clarification (Logs)](https://specs.optimism.io/protocol/derivation.html#on-future-proof-transaction-log-derivation)
+- [Specs clarification (L1 Retrieval)](https://specs.optimism.io/protocol/derivation.html#l1-retrieval)
+
+## Failure Modes and Recovery Paths
+
+### FM1: Incidental breaking changes from upstream geth
+
+- **Description:** Incidental changes from merging upstream geth cause unexpected problems. Example: database schema change can break op-geth nodes on rollout.
+- **Risk Assessment:** Medium severity, Medium Likelihood
+- **Mitigations:**:
+  1. Our usual staggered rollout and baking processes should catch most problems early.
+  2. Prepare appropriate runbook(s) detailing how nodes may need to be carefully migrated.
+- **Detection:** Alerts from our own infrastructure and/or that of partners'.
+- **Recovery Path(s)**: Follow the above runbook.
+
+### FM2: Consensus bug (client software)
+
+- **Description:** L2 EL/CL clients are not able to parse and/or validate blocks when Pectra goes live on L1, those nodes may halt. This could happen a) if operators do not update their nodes to a suitable release or release candidate before Pectra activates on L1 or b) no such release is available in time or c) there is a bug in such releases.
+- **Risk Assessment:** High severity, Low Likelihood
+- **Mitigations:**:
```

Comment:
Worth adding as mitigations for FM2 and FM3 that we perform Kurtosis testing with a L1 that has Pectra activated. Without it, all the end-2-end testing we do right now is mostly useless, because all L1 testnets don't have Pectra activated yet. Also, Sepolia will activate Pectra well before Mainnet, so it will provide the final confirmation for mainnet.